toolbox/bulk_processing/validators.py

```diff
@@ -123,3 +131,32 @@ def validate(expected_capacity, **kwargs):
                 f'CE Unit Expected Capacity "{expected_capacity}" cannot be null, blank or zero')
 
     return validate
+
+
+def ce_e_has_expected_capacity():
+    def validate(expected_capacity, **kwargs):
+        if kwargs['row']['ADDRESS_TYPE'] == 'CE' and kwargs['row']['ADDRESS_LEVEL'] == 'E' and (
+                kwargs['row']['TREATMENT_CODE'] not in {'CE_LDCEE', 'CE_LDCEW'}) and (
+                not expected_capacity.isdigit() or int(expected_capacity) == 0):
+            raise Invalid(
+                f'CE Estab Expected Capacity "{expected_capacity}" with Treatment Code '
+                f'{kwargs["row"]["TREATMENT_CODE"]} other than CE_LDCEE/CE_LDCEW cannot be null, blank or zero')
+
+    return validate
+
+
+def alphanumeric_postcode():
+    def validate(postcode, **_kwargs):
+        stripped_postcode = postcode.replace(" ", "")
+        if not stripped_postcode.isalnum():
+            raise Invalid(f'Postcode "{postcode}" is non alphanumeric')
+
+    return validate
+
+
+def latitude_longitude_range():
+    def validate(value, **_kwargs):
+        if not -180 <= float(value) <= 180:
```

Comment:
This needs to catch a ValueError and raise it as an Invalid, as is if the value can't be converted to float it kills the script.